src/developer-quickstart.md

```diff
@@ -0,0 +1,441 @@
+
+```console
+address fuel160ek8t7fzz89wzl595yz0rjrgj3xezjp6pujxzt2chn70jrdylus5apcuq
+private key 0x719fb4da652f2bd4ad25ce04f4c2e491926605b40e5475a80551be68d57e0fcb
+```
+
+Copy the private key, you will need this later to set it as a string value for a variable `WALLET_SECRET` in your `App.tsx` file. More on that below.
+
+First, take the address of your wallet and use it to get some coins from [the testnet faucet](https://faucet-beta-1.fuel.network/).
+
+Now you're ready to build and ship ⛽
+
+> Note: The team is working to simplify the process of creating a wallet, and eliminate the need to create a wallet twice. Keep an eye out for these updates.
+
+### Modify the App
+
+Inside the `frontend` folder let's add code that interacts with our contract.
+Read the comments to help you understand the App parts.
+
+Change the file `counter-dapp/frontend/src/App.tsx` to:
+
+```ts
+import React, { useEffect, useState } from "react";
+import { Wallet } from "fuels";
+import "./App.css";
+// Import the contract factory -- you can find the name in index.ts.
+// You can also do command + space and the compiler will suggest the correct name.
+import { CounterDappAbi__factory } from "./contracts";
+// The address of the contract deployed the Fuel testnet
+const CONTRACT_ID = "0xaf94c0a707756caae667ee43ca18bace441b25998c668010192444a19674dc4f";
+//the private key from createWallet.js
+const WALLET_SECRET = "0x719fb4da652f2bd4ad25ce04f4c2e491926605b40e5475a80551be68d57e0fcb";
+//pass in the testnet node
+const wallet = new Wallet(WALLET_SECRET, "https://node-beta-1.fuel.network/graphql");
+// Connects out Contract instance to the deployed contract
+// address using the given wallet.
+const contract = CounterDappAbi__factory.connect(CONTRACT_ID, wallet);
+
+function App() {
+  const [counter, setCounter] = useState(0);
+
+  useEffect(() => {
+    async function main() {
+      // Executes the counter function to query the current contract state
+      // the `.get()` is read-only, because of this it don't expand coins.
+      const { value } = await contract.functions.count().get();
+      setCounter(Number(value));
+    }
+    main();
+  }, []);
+
+  async function increment() {
+    // Creates a transactions to call the increment function passing the amount
+    // we want to increment, because it creates a TX and updates the contract state
+    // this requires the wallet to have enough coins to cover the costs and also
+    // to sign the Transaction
+    await contract.functions.increment().txParams({gasPrice:1}).call();
+    const { value } = await contract.functions.count().get();
+    setCounter(Number(value));
+  }
+
+  return (
+    <div className="App">
+      <header className="App-header">
+        <p>Counter: {counter}</p>
+        <button onClick={increment}>Increment</button>
+      </header>
+    </div>
+  );
+}
+
+export default App;
```

Comment:
As the test net is not as fast as local development, we could add a loader state on the button.

```suggestion
function App() {
  const [counter, setCounter] = useState(0);
  const [loading, setLoading] = useState(false);

  useEffect(() => {
    async function main() {
      // Executes the counter function to query the current contract state
      // the `.get()` is read-only, because of this it don't expand coins.
      const { value } = await contract.functions.count().get();
      setCounter(Number(value));
    }
    main();
  }, []);

  async function increment() {
    setLoading(true);
    // Creates a transactions to call the increment function passing the amount
    // we want to increment, because it creates a TX and updates the contract state
    // this requires the wallet to have enough coins to cover the costs and also
    // to sign the Transaction
    try {
      await contract.functions.increment().txParams({gasPrice:1}).call();
      const { value } = await contract.functions.count().get();
      setCounter(Number(value));
    } finally {
      setLoading(false);
    }
  }

  return (
    <div className="App">
      <header className="App-header">
        <p>Counter: {counter}</p>
        <button
          disabled={loading}
          onClick={increment}>
          {loading ? "Incrementing..." : "Increment"}
        </button>
      </header>
    </div>
  );
}

export default App;
```